### ISSUE TITLE
Not so important issue, but might be useful to take a look

### ISSUE THREAD

**Issue by jasonrichdarmawan** [cvenhoff/steering-thinking-llms#1]
Disclaimer: I am extending this work, my intention is to understand how this work works. If you are interested, this is the [repository](https://github.com/jasonrichdarmawan/steering-vectors-on-recurrent-model/blob/main/experiments/analyze_steering_effect_per_layer/main.py)

=== === Issue 1

Wouldn't the following code become a problem when the batch size is larger than 1?

https://github.com/cvenhoff/steering-thinking-llms/blob/83fc94a7c0afd9d96ca418897d8734a8b94ce848/vector-layer-attribution/analyze_layer_effects.py#L82

=== === Issue 2

Suppose a text `["example-testing"]1+1 = 2["end-section"]["backtracking"]Wait..., what if bla bla bla["end-section"]`. If `value = compute_kl_divergence_metric(logits[0, start-1:start].mean(dim=0))`. Then, does it mean we are computing the KL divergence between detached logits and logits of `1+1 = 2`?

https://github.com/cvenhoff/steering-thinking-llms/blob/83fc94a7c0afd9d96ca418897d8734a8b94ce848/vector-layer-attribution/analyze_layer_effects.py#L70

=== === Issue 3

Isn't this code misleading? `.mean(dim=0)` is averaging one token, so the output is the token itself.

https://github.com/cvenhoff/steering-thinking-llms/blob/83fc94a7c0afd9d96ca418897d8734a8b94ce848/vector-layer-attribution/analyze_layer_effects.py#L70

**Comment by jasonrichdarmawan** [comment]
Edit for Issue 2

Since the input_ids are processed, each token will predict the next token. In other words position `start-1` is that `1 + 1 = 2` predicting the next token.
